### PR TITLE
fix(snaps): Adjust alignment of custom UI links

### PR DESCRIPTION
**Pull request by GuillaumeRx**
## **Description**

This PR fixes the alignment issues of the custom UI link.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27957?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

```ts
import {
  Address,
  Box,
  Button,
  Container,
  Footer,
  Link,
  Row,
  Text,
  type SnapComponent,
} from '@metamask/snaps-sdk/jsx';

/**
 * A custom dialog component.
 *
 * @returns The custom dialog component.
 */
export const CustomDialog: SnapComponent = () => (
  <Container>
    <Box>
      <Row label="Test">
        <Link href="https://metamask.io">
          <Address address="0xc366063eA70eBD30316062C9C82445591c772870" />
        </Link>
      </Row>
      <Link href="https://metamask.io">This is a link</Link>
      <Text>
        This is a long text that should wrap around and take up multiple lines
        to test the layout of the dialog with a{' '}
        <Link href="https://metamask.io">
          very very very very long link that should wrap on multiple lines, we
          will see
        </Link>{' '}
        in the middle.
      </Text>
      <Link href="https://metamask.io">
        <Icon name="search" />
      </Link>
      <Link href="https://metamask.io">
        <Address address="0xc366063eA70eBD30316062C9C82445591c772870" />
      </Link>
    </Box>
    <Footer>
      <Button name="cancel">Cancel</Button>
      <Button name="confirm">Confirm</Button>
    </Footer>
  </Container>
);
```

### **Before**

Screenshot from 2024-10-21 12-34-40

### **After**

Screenshot from 2024-10-21 15-06-57

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/snaps/snap-ui-link/snap-ui-link.js [hunk 1]
```diff
@@ -52,6 +56,11 @@ export const SnapUILink = ({ href, children }) => {
         size={ButtonLinkSize.Inherit}
         display={Display.Inline}
         className="snap-ui-link"
+        textProps={{
+          display: Display.InlineFlex,
```
[the remark this answers is not in the document]
Comment (GuillaumeRx) [reply ui/components/app/snaps/snap-ui-link/snap-ui-link.js:60]: Yup :)

Screenshot from 2024-10-18 14-26-21

[the remark this answers is not in the document]
Comment (GuillaumeRx) [reply ui/components/app/snaps/snap-ui-link/snap-ui-link.js:64]: Screenshot from 2024-10-18 16-47-46

**Review by david0xd — DISMISSED** [review]

**Review by Mrtenz — DISMISSED** [review]

File: ui/components/app/snaps/snap-ui-renderer/index.scss [hunk 2]
```diff
@@ -48,10 +48,6 @@
 
   &__panel {
     gap: 8px;
-
-    .mm-icon--size-inherit {
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-renderer/index.scss:52]: I would imagine that this breaks this use-case: https://github.com/MetaMask/metamask-extension/pull/26904

Comment (GuillaumeRx) [reply ui/components/app/snaps/snap-ui-renderer/index.scss:52]: Let me check

Comment (GuillaumeRx) [reply ui/components/app/snaps/snap-ui-renderer/index.scss:52]: Fixed in bb3cc6e27be8f70cd0968c536848c0a2a0cab1a0

Screenshot from 2024-10-21 17-03-34

**Review by FrederikBolding — APPROVED** [review]

**Review by david0xd — APPROVED** [review]

**Pull request merged**